dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyConnectionClient.java

```diff
@@ -361,21 +360,14 @@ public void operationComplete(ChannelFuture future) {
                         "%s is reconnecting, attempt=%d cause=%s",
                         connectionClient, 0, future.cause().getMessage()));
             }
-            final EventLoop loop = future.channel().eventLoop();
-            loop.schedule(
-                    () -> {
-                        try {
-                            connectionClient.doConnect();
-                        } catch (RemotingException e) {
-                            LOGGER.error(
-                                    TRANSPORT_FAILED_RECONNECT,
-                                    "",
-                                    "",
-                                    "Failed to connect to server: " + getConnectAddress());
-                        }
-                    },
-                    1L,
-                    TimeUnit.SECONDS);
+            executor.submit(() -> {
```

Comment:
Also, please create a new isolated executor to reconnect

Reply:
Thank you very much.
I resolved the error by defining the ExecutorService within the method. However, a new error has occurred, so I will check it. 
Considering the current modification, instantiating it inside the method might be costly. Would it be better to have an instance of the ExecutorRepository class as a field in the NettyConnectionClient class?

Reply:
Yes, you're right. We need to maintain this executor in`ExecutorRepository`